packages/strapi-design-system/src/Accordion/Accordion.tsx

```diff
@@ -53,95 +59,87 @@ const AccordionWrapper = styled(Box)`
   }
 `;
 
-export const Accordion = ({
-  children,
-  expanded,
-  id,
-  size,
-  variant,
-  disabled,
-  error,
-  hasErrorMessage,
-  onToggle,
-  toggle,
-}) => {
-  const generatedId = useId('accordion', id);
-
-  return (
-    <AccordionContext.Provider value={{ expanded, onToggle, toggle, id: generatedId, size, variant, disabled }}>
-      <AccordionWrapper
-        data-strapi-expanded={expanded}
-        disabled={disabled}
-        aria-disabled={disabled}
-        expanded={expanded}
-        hasRadius
-        variant={variant}
-        error={error}
-      >
-        {children}
-      </AccordionWrapper>
-      {error && hasErrorMessage && (
-        <Box paddingTop={1}>
-          <Typography variant="pi" textColor="danger600">
-            {error}
-          </Typography>
-        </Box>
-      )}
-    </AccordionContext.Provider>
-  );
-};
+export type AccordionSize = 'S' | 'M';
+export type AccordionVariant = 'primary' | 'secondary';
 
-Accordion.defaultProps = {
-  disabled: false,
-  error: undefined,
-  expanded: false,
-  hasErrorMessage: true,
-  id: undefined,
-  toggle: undefined,
-  size: 'M',
-  variant: 'primary',
-  onToggle: undefined,
-};
-
-Accordion.propTypes = {
-  children: PropTypes.node.isRequired,
+export interface AccordionProps {
+  children: React.ReactNode;
   /**
    * If `true`, the accordion will be disabled.
    */
-  disabled: PropTypes.bool,
+  disabled?: boolean;
   /**
    * If defined, will add a border (borderColor: `danger600`) and display the error message below the component.
    */
-  error: PropTypes.string,
+  error?: string;
   /**
    * If `true`, an expanded Accordion will be rendered.
    */
-  expanded: PropTypes.bool,
+  expanded?: boolean;
   /**
    * If `false`, the error message won't show.
    * If the `Accordion` is as child of an `AccordionGroup`, this prop will be set to `false` automatically.
    * The error message of the `AccordionGroup` will be shown below the group instead of the Accordion itself.
    */
-  hasErrorMessage: PropTypes.bool,
+  hasErrorMessage?: boolean;
   /**
    * The id of the component.
    */
-  id: PropTypes.string,
+  id?: string;
```

Comment:
I know it wasn't set to required before, but isn't the id required in fact? Otherwise there would be conflicts when using multiple accordions at the same time? 🤔

Reply:
No the way the genId function works, we should be fine.